flake.nix

```
@@ -88,15 +88,20 @@
               compiler
               # TODO
               # Do we want to enable any `crossPlatforms` here?
-              ((infernoFor compiler pkgs).flake { })
+              ((infernoFor compiler pkgs { }).flake { })
+            ) ++
+          lib.lists.forEach [ defaultCompiler ]
+            (compiler: lib.attrsets.nameValuePair
+              (compiler + "-prof")
+              ((infernoFor compiler pkgs { profiling = true; ghcOptions = [ "-eventlog" ]; }).flake { })
             )
         );
 
       # This should be parameterized by the `pkgs` used to build the project. We
       # want users who get packages from our `overlays.default` to be able to
       # use their own `nixpkgs` without having to instantiate ours as well
       # (which would happen if we just use `self.packages` directly in the overlay)
-      infernoFor = compiler: pkgs: pkgs.haskell-nix.cabalProject {
+      infernoFor = compiler: pkgs: { ghcOptions ? [ ], profiling ? false }: pkgs.haskell-nix.cabalProject {
```

I think it would be best to make all of the arguments part of the attribute set now 
```suggestion
      infernoFor = { compiler, pkgs, ghcOptions ? [ ], profiling ? false }: pkgs.haskell-nix.cabalProject {
```